framework/src/meshgenerators/TransformGenerator.C

```diff
@@ -84,5 +84,6 @@ TransformGenerator::generate()
       break;
   }
 
+  mesh->set_isnt_prepared();
```

Comment:
why would that not be prepared? rotations, translations and scaling dont change neighbors?

Reply:
It invalidates the point locator and it can potentially make it so the spatial dimension of the mesh is incorrect (see #23923)